pkg/admission/machines.go

```diff
@@ -92,9 +92,9 @@ func (ad *admissionData) defaultAndValidateMachineSpec(spec *clusterv1alpha1.Mac
 		return fmt.Errorf("failed to get cloud provider %q: %v", providerConfig.CloudProvider, err)
 	}
 
-	// Verify operating system
-	if _, err := userdata.ForOS(providerConfig.OperatingSystem); err != nil {
-		return fmt.Errorf("failed to get OS '%s': %v", providerConfig.OperatingSystem, err)
+	// Verify operating system.
+	if !manager.Supports(providerConfig.OperatingSystem) {
```

Comment:
?

Reply:
`Supports()` doesn't care for supported OSses but if the `init()` has found a plugin for the given OS. Only a convenient wrapper instead of returning `ErrPluginNotFound`.  So `Supports()` is the wrong name, yes. Better would be `Found()`.